### DISCUSSION TITLE
Dados CPTEC

### DISCUSSION THREAD

**Discussion by RodriAndreotti** [BrasilAPI/BrasilAPI#172] [Ideas]
Boa noite, pessoal!

Hoje comecei a criar alguns endpoints para trazer os dados do [CPTEC](http://servicos.cptec.inpe.br/XML/), tentarei ajudar na issue #88, estes dados dependem de um código de cidade fornecido por eles, e foi justamente por este endpoint que comecei.

Falta somente uma rota para concluir a parte das cidades, gostaria da opinião de vocês, um dos endpoints faz a pesquisa de cidades pelo nome, e, se tratando de uma pesquisa, pode ocorrer de uma cidade retornar mais de uma ocorrência, pensei em fazer da seguinte forma:

- Caso o webservice deles retorne somente um resultado, retornar para o usuário final um objeto cidade
- Caso a resposta seja mais de uma cidade, ai sim retornar um array de objetos. (até o momento tudo é retornado como um array, mesmo que tenha somente uma posição)

Gostaria da opinião de vocês sobre este retorno, o que acham? 
Retorno sempre um array, mesmo com uma posição apenas, e mantenho o padrão do retorno? 
Ou retorno um objeto, quando o array tiver apenas uma posição?

Vlw!

**Comment by jlucas577** [comment]
Acredito que seja melhor retornar um array em todos os cenários.

**Reply by RodriAndreotti** [reply]
Estava pensando nisso também.

**Comment by RodriAndreotti** [comment]
Aproveitando, vocês acham legal traduzir os nomes das propriedades retornadas pelo CPTEC para inglês e tornar algo mais globalizado?

**Reply by rfoel** [reply]
Seguindo o padrão da resposta de CEP por exemplo eu diria que sim, deixar em inglês faz sentido.

**Reply by RodriAndreotti** [reply]
Eu pensei nisso também, pois ficaria mais padronizado, porém achei interessante perguntar, pois como é um API voltado para o público brasileiro, também faria sentido utilizar as propriedades em PT.
Minha preferência seria pelos nomes em inglês, mas como ainda não chegamos a um consenso sobre isso, é sempre melhor perguntar... rs

Vlw pela opinião

**Comment by RodriAndreotti** [comment]
Mais uma opinião que preciso dos senhores, pessoal!

Neste momento estou trabalhando no endpoint de previsão do tempo do CPTEC, e me surgiu a seguinte situação:
O CPTEC me retorna algumas informações da cidade junto a previsão, tenho como retornar desta forma:
![image](https://user-images.githubusercontent.com/9449048/102519385-2cadc580-4071-11eb-90e8-34403101945e.png)

Ou dessa forma:
![image](https://user-images.githubusercontent.com/9449048/102519469-45b67680-4071-11eb-9195-b184b3af525f.png)

Na opinião de vocês é melhor (e mais conciso) o retorno completo ou somente o array com as previsões?

**Reply by jlucas577-mob** [reply]
Acredito que o retorno completo é melhor

**Reply by RodriAndreotti** [reply]
Vlw pela opinião, PR feito!

**Reply by gustavo-oliveira-rosa** [reply]
Adorei a iniciativa! Também acho que o melhor a se fazer seja o envio completo dos dados!

**Reply by RodriAndreotti** [reply]
Vlw, criei já o Pull Request neste formato que sugeriu mesmo, já está na lista para validação... rs
